### PR TITLE
Prefer UINT_MAX instead of uint overflow.

### PR DESCRIPTION
**Pull request by RazrFalcon**
Also, prefer HB_FEATURE_GLOBAL_START and HB_FEATURE_GLOBAL_END.

### CODE REVIEW

File: src/hb-buffer.cc [hunk 1]
```diff
@@ -1738,7 +1738,7 @@ hb_buffer_add_codepoints (hb_buffer_t          *buffer,
  * @buffer: an #hb_buffer_t.
  * @source: source #hb_buffer_t.
  * @start: start index into source buffer to copy.  Use 0 to copy from start of buffer.
- * @end: end index into source buffer to copy.  Use (unsigned int) -1 to copy to end of buffer.
+ * @end: end index into source buffer to copy.  Use UINT_MAX to copy to end of buffer.
```
Comment (behdad) [inline src/hb-buffer.cc]: This should say `HB_FEATURE_GLOBAL_END`.

File: src/hb-common.h [hunk 2]
```diff
@@ -423,7 +423,7 @@ typedef void (*hb_destroy_func_t) (void *user_data);
  *
  * Since: 2.0.0
  */
-#define HB_FEATURE_GLOBAL_END	((unsigned int) -1)
+#define HB_FEATURE_GLOBAL_END	UINT_MAX
```
Comment (behdad) [inline src/hb-common.h]: I prefer not to rely on `UINT_MAX` in the public headers.

**Comment by RazrFalcon** [comment]
Well, I don't mind. I can rollback `UINT_MAX` and keep only `HB_FEATURE_GLOBAL_*`.

Personally, I think that `UINT_MAX` is just easier to read. `(unsigned int)-1` look like a weird magic.

**Comment by behdad** [comment]
Is okay to keep. Thanks. Just the two issues I pointed out in review.

**Comment by RazrFalcon** [comment]
Done.

File: src/hb-buffer.cc [hunk 3]
```diff
@@ -1738,7 +1738,7 @@ hb_buffer_add_codepoints (hb_buffer_t          *buffer,
  * @buffer: an #hb_buffer_t.
  * @source: source #hb_buffer_t.
  * @start: start index into source buffer to copy.  Use 0 to copy from start of buffer.
- * @end: end index into source buffer to copy.  Use (unsigned int) -1 to copy to end of buffer.
+ * @end: end index into source buffer to copy.  Use HB_FEATURE_GLOBAL_END to copy to end of buffer.
```
Comment (behdad) [inline src/hb-buffer.cc]: One more thing.  The ones in documentation I think should be written as `@HB_FEATURE_GLOBAL_END` for doc linking to work.

**Comment by RazrFalcon** [comment]
Done.

**Pull request merged**